exercises/joseph_ramsay/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+Command-line tool for counting unique words in a text file Usage:
+./WordCounter infile [outfile]
+ */
+
+#include <algorithm>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+/** Convert character to lower
+ *
+ * @param c The character to convert
+ * @return The converted character
+ */
+char charToLower(const char c) {
+  if (65 <= c && c <= 90)
+    return c + 32;
+  return c;
+}
+
+/** Convert a string to lower case
+ *
+ * @param s The string to convert
+ * @return The converted string
+ */
+string lower(const string& s) {
+  string t;
+  transform(s.begin(), s.end(), back_inserter(t), ::tolower);
+  return t;
+}
+
+/** Is a character whitespace or a dash?
+ *
+ * @param c The character to test
+ * @return True if the character is whitespace or hyphen, false otherise
+ */
+bool space(const char c) { return isspace(c) || c == '-'; }
+
+/** Is not the character not whitespace or a dash?
+ *
+ * @param c The character to test
+ * @return True if the character is not whitespace or hyphen, false otherwise
+ */
+bool notSpace(const char c) { return !space(c); }
+
+/** Split a string on whitespace and hyphens
+ *
+ * @param s The string to split
+ * @return A vector of the words in the string
+ */
+std::vector<string> split(const string& s) {
+  vector<string> words;
+  auto i = s.begin();
+
+  while (i != s.end()) { // While i is not at the end of s
+    // Put i at the start of the next word, and j at the end of it
+    i = find_if(i, s.end(), notSpace);
+    string::const_iterator j = find_if(i, s.end(), space);
+
+    if (i != s.end()) // Success, so add the word to the vector
+      words.push_back(string(i, j));
+    i = j;
+  }
+  return words;
+}
+
+/** Is a character not a lowercase character?
+ *
+ * @param c The character to test
+ * @return True if the character is any other than a-z, false otherwise
+ */
+bool notLowerAlNum(const char c) {
```

Comment:
You are indeed checking for numeric values, but numeric values were only an example. The requirement is to only not consider these items: `.,?'"!():`. If you have a `+` in your file it will also be discarded. You should check for `.,?'"!():` .